nmpolicy/internal/parser/parser.go

```diff
@@ -125,44 +134,59 @@ func (p *Parser) parsePath() (*ast.Node, error) {
 		if p.currentToken().Type == lexer.DOT {
 			p.nextToken()
 			if p.currentToken().Type == lexer.IDENTITY {
-				path := append(*operator.Path, p.parseIdentity())
-				operator.Path = &path
+				p.parseIdentity()
+			} else if p.currentToken().Type == lexer.NUMBER {
+				err := p.parseNumber()
+				if err != nil {
+					return &InvalidPathError{fmt.Sprintf("invalid number: %v", err)}
```

Comment:
The reason for the error ("invalid number") should be provided in the error itself, not here. Here we need to say that failed at this abstraction level (which I guess is embedded in the error type name).

Reply:
Doing proper error wrapping.